### PR TITLE
 Cleaned up cmake to be able to build on newer VS

### PR DESCRIPTION
**Pull request by davidstl**

### CODE REVIEW

File: autobuild/build_win20xx.bat [hunk 1]
```diff
@@ -11,15 +11,15 @@ cd build
 
 mkdir debug
 cd debug
-cmake -D NO_CPPREST_SDK:BOOL=1 -D SKIP_TESTS:BOOL=1 -D gtest_force_shared_crt:BOOL=1 -DCMAKE_BUILD_TYPE=Debug -G "NMake Makefiles" ..\..\..\..
```
Comment (davidstl) [inline autobuild/build_win20xx.bat]: Removed double negation...

File: tests/gtest-1.7.0/CMakeLists.txt [hunk 2]
```diff
@@ -14,7 +14,7 @@ option(BUILD_SHARED_LIBS "Build shared libraries (DLLs)." OFF)
 option(
   gtest_force_shared_crt
   "Use shared (DLL) run-time lib even when Google Test is built as static lib."
-  OFF)
+  ON)
```
Comment (davidstl) [inline tests/gtest-1.7.0/CMakeLists.txt]: This flag was not being set even passing it to the cmake argument on parent cmake... Forcing it there.
Anyway we were forcing it in 2008 and 2010 also.

File: autobuild/build_win20xx.bat (hunk 1 above)
Comment (kgeddie) [reply autobuild/build_win20xx.bat]: 😃

**Comment by kgeddie** [comment]
That's a big PR. I looked over it quickly. I didn't notice anything wrong.

**Comment by davidstl** [comment]
Tested VS2017 and OSX so far.

**Comment by davidstl** [comment]
Built iOS/TVOS also. Doesn't seem to be broken

**Comment by davidstl** [comment]
Build Linux and rand tests there too.

**Pull request merged**